compile/commons.py

```diff
@@ -0,0 +1,444 @@
+# Copyright (c) 2025 LATMOS (France, UMR 8190) and IGE (France, UMR 5001).
+#
+# License: BSD 3-clause "new" or "revised" license (BSD-3-Clause).
+
+"""Common resources for compilation scripts."""
+
+import os
+import functools
+import argparse
+import subprocess
+import json
+
+URL_GITHUB = "https://github.com"
+URL_GROUP_WRF = "%s/wrf-model" % URL_GITHUB
+URL_WPS = "%s/WPS.git" % URL_GROUP_WRF
+URL_GROUP_POLAR = "%s/WRF-Chem-Polar" % URL_GITHUB
+URL_WRFCHEMPOLAR = "%s/WRF-Chem-Polar.git" % URL_GROUP_POLAR
+
+
+class ConvertToBoolean(argparse.Action):
+    """Action to convert command-line arguments to booleans."""
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        """Convert command line option value to boolean.
+
+        See https://docs.python.org/3/library/argparse.html#action-classes for
+        more details about action classes and the corresponding API.
+
+        """
+        while option_string.startswith("-"):
+            option_string = option_string[1:]
+        option_string = option_string.replace("-", "_")
+        if values.lower() in ("true", "t", "yes", "y"):
+            values = True
+        elif values.lower() in ("false", "f", "no", "n"):
+            values = False
+        else:
+            raise ValueError('Could not convert "%s" to boolean.' % values)
+        setattr(namespace, option_string, values)
+
+
+@functools.lru_cache
+def identify_host_platform():
+    """Return the identity of the host platform.
+
+    Returns
+    -------
+    str
+        The identity of the host platform.
+
+    """
+    known_plateforms = {
+        "spirit1.ipsl.fr": "spirit",
+        "spirit2.ipsl.fr": "spirit",
+    }
+    nodename = os.uname().nodename
+    try:
+        platform = known_plateforms[nodename]
+    except KeyError:
+        raise NotImplementedError("Unknown host platform: %s." % nodename)
+    return platform
+
+
+def process_path(path):
+    """Return a unique absolute version of given path.
+
+    Parameters
+    ----------
+    path: str | None
+        The path to process.
+
+    Returns
+    -------
+    str | None
+        The unique and absolute version of given path (or None if given path is
+        empty or None)
+
+    """
+    if path is None or path.strip() == "":
+        return None
+    else:
+        return os.path.abspath(os.path.expanduser(path))
+
+
+@functools.lru_cache
+def path_of_repo():
+    """Return the path to the root of the repository that contains this file.
+
+    Returns
+    -------
+    str
+        The path to the root of the repository that contains this file.
+
+    """
+    path = os.path.dirname(__file__)
+    if path == "":
+        path = "."
+    return process_path(os.path.join(path, ".."))
```

Comment:
A potentially more generic version in standard python and without calling the git command:

```suggestion
def path_of_repo(file = None):
    """Return the path to the root of the repository that contains this file.

    Parameters
    ----------
    file: None or str
        The file or directory in which to search for the root path of the repository.
    
    Returns
    -------
    str
        The path to the root of the repository that contains this file.

    """
    parent = os.path.dirname(os.path.abspath(__file__ if file is None else file))
    if os.path.isdir(os.path.join(parent, ".git")):
        return parent

    return path_of_repo(parent)
```

Not sure if the default setting is the best.
The function seems to loop on itself if executed outside of a Git repo.